GTG/gtk/preferences.py

```diff
@@ -150,59 +136,68 @@ def _refresh_task_browser(self):
         task_tree.refresh_all()
 
     def activate(self):
-        """ Activate the preferences dialog."""
+        """ Activate the preferences window."""
         self._refresh_preferences_store()
-        self.dialog.show_all()
+        self.window.show_all()
+
+    def on_valid_check(self, widget):
+        try:
+            input_time = self.refresh_time.get_text()
+            self.timer.parse_time(input_time).strftime("%H:%M")
+            color = None
+
+        except (ValueError, TypeError):
+            color = self.INVALID_COLOR
+
+        self.refresh_time.modify_fg(Gtk.StateFlags.NORMAL, color)
 
-    def valid_check(self, widget):
+    def on_leave_entry(self, widget, data=None):
         try:
-            self.timer.set_configuration(self.refresh_hour.get_text(),
-                                         self.refresh_mins.get_text())
+            input_time = self.refresh_time.get_text()
+            correct_time = self.timer.parse_time(input_time)
+            self.timer.set_configuration(correct_time)
             color = None
+
         except (ValueError, TypeError):
-            color = self.color_invalid
+            color = self.INVALID_COLOR
 
-        self.refresh_hour.modify_fg(Gtk.StateFlags.NORMAL, color)
-        self.refresh_mins.modify_fg(Gtk.StateFlags.NORMAL, color)
+        self.refresh_time.modify_fg(Gtk.StateFlags.NORMAL, color)
+        self._refresh_preferences_store()
 
     def on_close(self, widget, data=None):
         """ Close the preferences dialog."""
-        self.valid_check(widget)
+        self.on_leave_entry(widget)
```

Comment:
Is this really needed? The event in that case should be triggered automatically.